app/controllers/items_controller.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+
+class ItemsController < ApplicationController
+
+	def new
+		@item = Item.new
+	end
+
+	def create
+		@item = Item.new
+
+		url = item_params[:item_url]
+
+		if !validate_url(url)
+			flash[:error] = "Please enter a valid Ikea product URL or Article Number!"
+			render :new and return
+		end
+
+		create_item(@item, url, item_params)
```

Comment:
Instead of creating `@item` and then passing it into `create_item`, you could use the `create_item` method to modify your `item_params` hash.

The hash that would be outputted from this method would allow you to do this:

```
@item = Item.new(create_item)
```